specification/dsdl/conventions.tex

```diff
@@ -0,0 +1,115 @@
+\section{Conventions and recommendations}
+
+This section is dedicated to conventions and recommendations
+intended to help data type designers maintain a consistent style across the ecosystem
+and avoid some common pitfalls.
+Some of the conventions are mandatory to follow, others are optional.
+
+\subsection{Standard namespace}
+
+This specification defines a set of standard regulated DSDL data types located under
+the root namespace named ``\verb"uavcan"'' (section~\ref{sec:list_of_standard_data_types}).
+
+Vendor-specific, user-specific, or any other data types not defined by this specification
+must not be defined inside the standard root namespace.
+Vendor-specific or user-specific root namespaces must be used instead.
+
+\subsection{Naming recommendations}
+
+The DSDL naming recommendations follow those that are widely accepted in the general software development industry.
+
+\begin{itemize}
+    \item Namespaces and field attributes should be named in the \verb|snake_case|.
+    \item Constant attribute should be named in the \verb|SCREAMING_SNAKE_CASE|.
+    \item Data types (excluding their namespaces) should be named in the \verb|PascalCase|.
+    \item Names of message types should form a declarative phrase or a noun. For example,
+    \verb|BatteryStatus| or \verb|OutgoingPacket|.
+    \item Names of service types should form an imperative phrase or a verb. For example,
+    \verb|GetInfo| or \verb|HandleIncomingPacket|.
+    \item Avoid short names, unnecessary abbreviations, and uncommon acronyms.
+\end{itemize}
+
+\subsection{Comments}
+
+Every data type definition file should begin with a header comment that provides an exhaustive description
+of the data type, its purpose, semantics, usage patterns, any related data exchange patterns,
+assumptions, constraints, and all other information that may be necessary or generally useful for the usage of the
+data type definition.
+
+Every attribute of the data type definition, and especially every field attribute of it,
+should have an associated comment explaining the purpose of the attribute, its semantics, usage patterns,
+assumptions, constraints, and any other pertinent information.
+Exception applies to attributes supplied with sufficiently descriptive and unambiguous names.
+
+% Field comment placement https://forum.uavcan.org/t/dsdl-documentation-comments/407
+
+\subsection{Optional value representation}
+
+Data structures may include optional field attributes that are not necessarily always populated.
```

Comment:
```suggestion
Data structures may include optional field attributes that are not always populated.
```
```suggestion
Data structures may include optional field attributes that are not necessarily populated.
```

These suggestions aside; I think we mean to say something like "There is no built-in support for optional field attributes but here's how you can fake it." If so then the former is normative and the latter is non-normative.